### PR TITLE
Workaround for missing _mm_loadu_si64 on old compilers

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
Fixes https://github.com/Tarsnap/tarsnap/issues/473.

File: crypto/crypto_aesctr_aesni.c [hunk 1]
```diff
@@ -60,7 +64,11 @@ crypto_aesctr_aesni_stream_wholeblocks(struct crypto_aesctr * stream,
 		be64enc(block_counter_be_arr, block_counter);
 
 		/* Encrypt the cipherblock. */
+#ifdef NO_LOADU_SI64
+		bufsse = _mm_loadu_si128((const __m128i *)block_counter_be_arr);
```
Comment (cperciva) [inline crypto/crypto_aesctr_aesni.c:68]: This performs a 128-bit load from an 8-byte array, which seems a bit optimistic.

**Comment by cperciva** [comment]
I'm not entirely certain this is worth complicating `crypto_aesctr_aesni` for.  We could fix the build breakage by using `_mm_loadu_si64` in the compiler support detection -- people with broken compilers then wouldn't get the optimized version, but they're not going to get a fully optimized version anyway and they should upgrade to a working compiler.

File: crypto/crypto_aesctr_aesni.c (hunk 1 above)
Comment (gperciva) [reply crypto/crypto_aesctr_aesni.c:68]: ah right, I was only thinking about stream->pblk.

**Comment by gperciva** [comment]
> I'm not entirely certain this is worth complicating crypto_aesctr_aesni for.

I agree that the current PR is flawed, but I think we should have something.  The previous releases of spiped and tarsnap had working AESNI.

Now, one possible fix would be to still use `crypto_aes_aesni.c` but not `crypto_aesctr_aesni.c` (which is essentially what the previous releases had).  But in some ways that ends up being a more complicated solution, because in that case we'd need cpusupport to detect 1) can we use AESNI, and 2) does the compiler recognize `_mm_loadu_si64`.  (Which brings us one step closer to the dreaded `./configure`.)

**Comment by gperciva** [comment]
Better fix -- we can use `_mm_load_sd()` to do exactly the same thing as `_mm_loadu_si64()`, albeit with a cast.  I've hidden this inside a `load_si64()` shim.  The compiler ends up using `movq` in both cases.

This allows us to (mostly) keep the recent optimizations.  (gcc7 is "only" 1100 MB/s, rather than 1600)

### Compare versions of gcc

compilers

**Comment by gperciva** [comment]
Done.

**Comment by cperciva** [comment]
Commit message needs updating for the changed macro name.

**Comment by gperciva** [comment]
Oops, fixed.

**Pull request merged**